cmd/lnmuxd/run.go

```diff
@@ -48,6 +49,27 @@ func runAction(c *cli.Context) error {
 		return err
 	}
 
+	group, ctx := errgroup.WithContext(context.Background())
+
+	// Initialise instrumentation server
+	instServer := initInstrumentationServer(cfg.InstrumentationAddress)
+	group.Go(func() error {
```

Comment:
Is there a reason to use errGroup instead if waitGroup?

I stick with errGroup, but it is less flexible here.

Here we have a problem if we shutdown the process for an instance that waits for the lease.
Indeed, if we shutdown the process during `initDistributedLock`, we will not shutdown gracefully the metrics server.

2 solutions:

- use waitGroup for this one (but it is not really consistent w/ errGroup after
- we don't care, closing gracefully the instrumentation server is not really critical

Reply:
I lean toward #2, we don't really care if the instrumentation server closes gracefully.

Reply:
We used waitgroup before, but it can get complicated with waiting for all the goroutine results and keeping track of what to terminate.

If `initDistrbutedLock` returns an error, couldn't you cancel the context and `group.Wait()`?

Reply:
Oh, I see this is what @aakselrod-nydig suggests, but then with an added wait

Reply:
Yeah we could do that,

I found this strange to do this only here. 
Why not doing this for `mux,err := lnmux.New(&lnmux.MuxConfig{...})`, etc... 

But I suppose, as I said, this is the most likely of unlikely scenarios! So it could make sense!

Reply:
I see. How about `defer group.Wait()` ?

Reply:
Was thinking of this indeed, but we want to return the value of `group.Wait()` (if any error)
And we can't do this in a defer.

But we can have two group.Wait(), one defer and the other one at the end at the function. A bit strange but it is okay for me!

(as a note: group.Wait() is not idempotent)

Reply:
You can return that `group.Wait` result if you make it a named return parameter I think.

Reply:
Indeed, it is less readable but It avoids to use 2 group.Wait()

Reply:
Or indeed just do something special only for the admin server and don't include it in the global errgroup.